lib/ip-locator.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+const Ipinfo = require('ipinfo')
+const RequestIp = require('request-ip')
+
+class IpLocator {
+  constructor (options) {
+    const { enabledByDefault = true } = options
+    this.enabledByDefault = enabledByDefault
+  }
+
+  /**
+   * @function setting
+   * @description Return the setting for the hapi-geo-locate by taking the request as a parameter and return the setting
+   * @param {Object} request
+   * @returns {Object}
+   */
+  setting (request) {
+    return request.route.settings.plugins['hapi-geo-locate'] || {}
+  }
+
+  /**
+   * @async
+   * @function handle
+   * @description This is the main function to handle the request
+   * @param {Object} request
+   * @param {Object} h
+   */
+  async handle (request, h) {
+    if (this.isDisabledFor(request)) {
+      return h.continue
+    }
+    request.location = await this.locate(request)
+    return h.continue
+  }
+
+  /**
+   * @function isDisabledFor
+   * @param {Object} request
+   * @returns Boolean
+   */
+  isDisabledFor (request) {
+    const settings = this.setting(request)
```

Comment:
This method body is a copy of the old code. Let's refactor the code to be more maintainable and readable. I suggest to add another method method:

```suggestion
    return !this.isEnabledFor(request)
```

And the implementation of `isEnabledFor(request)` could look like this:

```js
const { enabled = this.enabledByDefault } = this.routeConfig(request)

return enabled
```